File: B3WM.Shared/Models/Defaults.cs
namespace B3WM.Shared.Models
{
    public static class Defaults
    {
        public const string Url = "https://localhost:5002/api/datahub";
        public static readonly int[] TimeFrames = [1, 2, 5, 15, 30, 60];

        public static class Symbols
        {
            public const string WINFUT = "WINFUT";
            public const string WDOFUT = "WDOFUT";
        }

        public static class WINFUT
        {
            public const int ThresholdBubbleSize = 250;
            public const double MinDistanceUpdateBorder = 250;
            public const double TickSize = 5.0;
            public const double PointValue = 1.0;
        }

        public static class WDOFUT
        {
            public const int ThresholdBubbleSize = 500;
            public const double MinDistanceUpdateBorder = 2.5;
            public const double TickSize = 0.5;
            public const double PointValue = 10.0;
        }

        public static class Backtest
        {
            public const int SmartEntryThreshold = 500;
            public const int SmartExitThreshold = 1000;
            public const double SmartVolumePct = 0.3;
            public const double SmartStructureBufferPct = 0.1;
        }

        public static double GetPointValue(string symbol) => symbol switch
        {
            Symbols.WINFUT => WINFUT.PointValue,
            Symbols.WDOFUT => WDOFUT.PointValue,
            _ => 1.0
        };

        public static double GetTickSize(string symbol) => symbol switch
        {
            Symbols.WINFUT => WINFUT.TickSize,
            Symbols.WDOFUT => WDOFUT.TickSize,
            _ => 1.0
        };

        public static double GetMinDistance(string symbol) => symbol switch
        {
            Symbols.WINFUT => WINFUT.MinDistanceUpdateBorder,
            Symbols.WDOFUT => WDOFUT.MinDistanceUpdateBorder,
            _ => 250
        };
    }
}

File: B3WM.Flutter/lib/models/defaults.dart
class Defaults {
  static const String url = 'https://localhost:5002/api/datahub';
  static const List<int> timeFrames = [1, 2, 5, 15, 30, 60, 1440];

  static String timeFrameLabel(int tf) => tf == 1440 ? '1D' : '$tf';

  static const String symbolWinfut = 'WINFUT';
  static const String symbolWdofut = 'WDOFUT';

  static int thresholdBubbleSize(String symbol) {
    switch (symbol) {
      case 'WINFUT':
        return 250;
      case 'WDOFUT':
        return 500;
      default:
        return 250;
    }
  }

  static double minDistanceUpdateBorder(String symbol) {
    switch (symbol) {
      case 'WINFUT':
        return 250;
      case 'WDOFUT':
        return 4.5;
      default:
        return 250;
    }
  }

  static double tickSize(String symbol) {
    switch (symbol) {
      case 'WINFUT':
        return 5.0;
      case 'WDOFUT':
        return 0.5;
      default:
        return 1.0;
    }
  }

  static double pointValue(String symbol) {
    switch (symbol) {
      case 'WINFUT':
        return 1.0;
      case 'WDOFUT':
        return 10.0;
      default:
        return 1.0;
    }
  }

  static double structureRangeUpdStep(String symbol) {
    switch (symbol) {
      case 'WDOFUT':
        return 0.5;
      default:
        return 5.0;
    }
  }

  static double structureRangeUpdMax(String symbol) {
    switch (symbol) {
      case 'WDOFUT':
        return 100;
      default:
        return 2000;
    }
  }

  static const double extremeNoiseSensitivity = 3.0;
  static const double extremeMinimumProminence = 0.15;

  static const double extremeNoiseSensitivityMin = 1.0;
  static const double extremeNoiseSensitivityMax = 10.0;
  static const double extremeNoiseSensitivityStep = 0.5;
  static const double extremeMinimumProminenceMin = 0.05;
  static const double extremeMinimumProminenceMax = 0.5;
  static const double extremeMinimumProminenceStep = 0.05;
}
